### ISSUE TITLE
Accessibility of config forms

### ISSUE THREAD

**Issue by LucyGwilliamAdmin** [open-sdg/open-sdg#1000]
Feedback from DAC re. site and indicator config forms:
Note: "I" is from tester not from me

I have had a look at the first URL (http://sdgdev-813006012.eu-west-1.elb.amazonaws.com/release-prep-and-config-forms/config/) this morning which I have used one hour for. I imagine that issues will be consistent across the two pages, but please let me know if you would like me to use another hour for the second page. I have recorded some comments which would likely be accessibility barriers and that I would expect to be identified if you were to have an expert review or a full audit. I imagine that fixing these would go a substantial way to making the pages fully accessible, and I didn’t notice anything else immediately obvious to myself as an issue within the high level look I was able to give it in the hours consultancy slot. However please do note that this has not been fully tested with all the technologies we would use for a full audit, and that other issues may still be present. I have recorded some issues below that I would flag.

1. Unlabelled form fields (A issue – name, role and value)
  
    ![image](https://user-images.githubusercontent.com/43404358/101171139-45f94f80-3637-11eb-8a7e-02e0c772bf62.png)

    Please note, I expanded some of the sections and ran axe on this which noted 128 form fields without programmatically associated labels. This means they have no accessible name to present their purpose to screen reader users or to allow voice activation users to reference them to their assistive technology by the visible label. 

2. Headings don’t reflect hierarchy or indicate sections or regions on the page (A issue – info and relationships)

    ![image](https://user-images.githubusercontent.com/43404358/101171252-67f2d200-3637-11eb-9d0d-e8a7deb160cf.png)

    This may be particularly confusing as there is a button which expands the ‘breadcrumbs section’. This triggers content to appear further down the page, but when this content is appears, because the headings are not subheadings of ‘breadcrumbs’ there is nothing to indicate to screen reader users that this is the newly triggered content which belongs to the visually indicated (by a grey background) ‘breadcrumbs section’. 

3. When the page is magnified, horizontal scrolling is induced due to the use of tables
    
    ![image](https://user-images.githubusercontent.com/43404358/101171357-89ec5480-3637-11eb-8ebc-21fe860d0e7b.png)

4. JAWS was not triggering the input type colour when I tested in Chrome – it seemed to only activate the link above. I tested JAWS with the same type of field on another site and it worked with Chrome, so I don’t believe this is a JAWS issue, but I would need to have a bit of a dig to try and work out why this isn’t recognised. The page wouldn’t load in IE for me. 

    ![image](https://user-images.githubusercontent.com/43404358/101171427-a8525000-3637-11eb-8710-bae31b799a3a.png)

5. None of these buttons are keyboard focusable (A issue keyboard). It is also noted that there is nothing for screen reader users to determine whether these are ‘on’ or ‘off’ (A info and relationships). 

    ![image](https://user-images.githubusercontent.com/43404358/101171602-e485b080-3637-11eb-82c6-ed169ecbc98b.png)

6. The lines/word counter update dynamically but are not relayed to screen reader users (AA Status messages). 
 
    ![image](https://user-images.githubusercontent.com/43404358/101171643-f6675380-3637-11eb-8d5e-f64cd92ef343.png)

7. There are links which open in a new window and do not inform users of this prior to activation 

8. Results update dynamically without being explicitly requested or informing users that the page has updated. I would recommend a submit button and a status message to inform users how many results have been returned.
 
    ![image](https://user-images.githubusercontent.com/43404358/101171692-0a12ba00-3638-11eb-8ca8-b58b6062cd1e.png)

9. There is an illogical heading structure has there are h3 headings which are not subheadings of a h2.

**Comment by brockfanning** [comment]
As I worried there are many accessibility problems, even with a 1-hour audit. I think there are some nice-to-haves which we could simply turn off, to avoid problems:

1. Table-format in some multiple-item settings: we can turn off "table" mode and just display the items vertically. It takes up a lot more vertical space but maybe that's OK since they start off collapsed.
2. The markdown editor buttons: We could simply turn off that markdown editor and have a plain text field. Or potentially see if other markdown editors might work better.
3. Color fields: we could turn off the special color field functionality and just have plain text fields.

I think this would help with some of the items. Others I may be able to code fixes for.

**Comment by brockfanning** [comment]
I think this one is on hold until a third-party library (json-editor) gets a new feature and release. More discussion in #1003.
